File: Python_Ruby/OOP_Variable/3.py
class Calculator(object):
    def __init__(self, v1, v2):
        self.v1 = v1
        self.v2 = v2

    def setV1(self, v):
        if(isinstance(v, int)):
            self.v1 = v
        else:
            print('You tried changing v1 by not integer')

    def getV1(self):
        return self.v1

    def add(self):
        return self.v1 + self.v2
    def subtract(self):
        return self.v1 - self.v2

c1 = Calculator(10, 10)
print(c1.add())
print(c1.subtract())

c1.setV1(20)
print(c1.add())

c1.setV1('one')
print(c1.add())

File: Python_Ruby/OOP_Variable/3.rb
class Calculator
    def initialize(v1, v2)
        @v1 = v1
        @v2 = v2
    end

    def setV1(v)
        if v.is_a?(Integer)
            @v1 = v
        else
            p 'you tried changing v1 by not integer'
        end
    end

    def add()
        return @v1 + @v2
    end

    def subtract()
        return @v1 - @v2
    end
end

c1 = Calculator.new(10, 10)
p c1.add()
p c1.subtract()

c1.setV1(20)
p c1.add()

c1.setV1('one')
p c1.add()